### PR TITLE
Create Kadanes.md

### LINKED ISSUE #571: [Feature Request]:  Add Kadanes Algorithm's Documentation
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

Greetings !!

Abhay here :) I would like to add Kadane's Algorithm Documentation for the website and also the code in 4 languages.
I hope this finds useful.

Thank You

### Use Case

Kadane's algorithm is used for optimizing the problem ,mainly for max subarray question

### Benefits

_No response_

### Add ScreenShots

_No response_

### Priority

Medium

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by Abhay182005dat**
Greetings !! 
This closes #571 

Sir I have succesfully completed my Documentation of Kadane's Algorithm !

Thank You :)

### CODE REVIEW

File: docs/dsa/Kadanes.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+---
+id: Algorithms
```
Comment (Ajay-Dhangar) [inline docs/dsa/Kadanes.md:2]: replace `id: Algorithms` to `id: kadanes-algorithm`

File: docs/dsa/Kadanes.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+title: Kadanes Algorithm
+sidebar_label: Algorithms
+sidebar_position: 2
+description: "Kadane's Algorithm efficiently finds the maximum sum contiguous subarray in a one-dimensional array. It's a dynamic programming technique named after its creator, Jay Kadane."
+---
+## Kadane's Algorithm:
+
+- **Purpose**: Finds the maximum sum of a contiguous subarray.
+- **Time Complexity**: O(n)
```
Comment (Ajay-Dhangar) [inline docs/dsa/Kadanes.md:11]: replace `O(n)` to `$O(n)$ and `O(1)` to `$O(1)$`

File: docs/dsa/Kadanes.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
+---
+id: Algorithms
+title: Kadanes Algorithm
+sidebar_label: Algorithms
```
Comment (Ajay-Dhangar) [inline docs/dsa/Kadanes.md:4]: replace `sidebar_label: Algorithms` to `sidebar_label: Kadanes Algorithm`

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update code based on suggestions

File: docs/dsa/Kadanes.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+   - The value of `maxSum` at the end of the iteration is the maximum sum of any contiguous subarray.
+### Code
+
+Input: arr = {-2,-3,4,-1,-2,1,5,-3}
+
+Output: 7
+
+Explanation: The subarray {4,-1, -2, 1, 5} has the largest sum 7.
+
+### Python
```
Comment (Ajay-Dhangar) [inline docs/dsa/Kadanes.md:37]: ### Bad practice

```md

    def max_sub_array(nums):
          max_sum = float('-inf')
          current_sum = 0

         for num in nums:
            # Add the current number to the current_sum
            current_sum += num
            # Update max_sum if current_sum is greater
            max_sum = max(max_sum, current_sum)
            # Reset current_sum to 0 if it drops below 0
            if current_sum < 0:
                current_sum = 0
            
         return max_sum
```

### Good Practice

````md
```py

    def max_sub_array(nums):
          max_sum = float('-inf')
          current_sum = 0

         for num in nums:
            # Add the current number to the current_sum
            current_sum += num
            # Update max_sum if current_sum is greater
            max_sum = max(max_sum, current_sum)
            # Reset current_sum to 0 if it drops below 0
            if current_sum < 0:
                current_sum = 0
            
         return max_sum
```
````

File: docs/dsa/Kadanes.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+            # Update max_sum if current_sum is greater
+            max_sum = max(max_sum, current_sum)
+            # Reset current_sum to 0 if it drops below 0
+            if current_sum < 0:
+                current_sum = 0
+            
+         return max_sum
+
+### C++
+    // Kadane's Algorithm to find the maximum sum of a contiguous subarray
```
Comment (Ajay-Dhangar) [inline docs/dsa/Kadanes.md:55]: ### bad practice

```md
   // Kadane's Algorithm to find the maximum sum of a contiguous subarray

    #include <iostream>
    #include <vector>
    #include <algorithm>
    #include <climits>

    int maxSubArray(const std::vector<int>& nums) {
    // Initialize max_sum to the lowest possible value and current_sum to 0
    int max_sum = INT_MIN;
    int current_sum = 0;

    for (int num : nums) {
        // Add the current number to the current_sum
        current_sum += num;
        // Update max_sum if current_sum is greater
        max_sum = std::max(max_sum, current_sum);
        // Reset current_sum to 0 if it drops below 0
        if (current_sum < 0) {
            current_sum = 0;
        }
    }

    return max_sum;
    }
```

### Good Practice

````md
```cpp
   // Kadane's Algorithm to find the maximum sum of a contiguous subarray

    #include <iostream>
    #include <vector>
    #include <algorithm>
    #include <climits>

    int maxSubArray(const std::vector<int>& nums) {
    // Initialize max_sum to the lowest possible value and current_sum to 0
    int max_sum = INT_MIN;
    int current_sum = 0;

    for (int num : nums) {
        // Add the current number to the current_sum
        current_sum += num;
        // Update max_sum if current_sum is greater
        max_sum = std::max(max_sum, current_sum);
        // Reset current_sum to 0 if it drops below 0
        if (current_sum < 0) {
            current_sum = 0;
        }
    }

    return max_sum;
    }
```
````

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update code based of best practice

**Comment by Abhay182005dat** [comment]
I guess this is done now sir...
Thank You again :)

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**